addons/website/static/src/snippets/s_searchbar/000.js

```diff
@@ -1,127 +1,103 @@
+            "displayImage": dataset.displayImage,
+            "displayDescription": dataset.displayDescription,
+            "displayExtraLink": dataset.displayExtraLink,
+            "displayDetail": dataset.displayDetail,
+            // Make it easy for customization to disable fuzzy matching on specific searchboxes
+            "allowFuzzy": !!dataset.noFuzzy,
         };
-        const form = this.$('.o_search_order_by').parents('form');
-        for (const field of form.find("input[type='hidden']")) {
-            this.options[field.name] = field.value;
+        const form = orderByEl.closest("form");
+        for (const fieldEl of form.querySelectorAll("input[type='hidden']")) {
+            this.options[fieldEl.name] = fieldEl.value;
         }
-        const action = form.attr('action') || window.location.pathname + window.location.search;
-        const [urlPath, urlParams] = action.split('?');
+        const action = form.getAttribute("action") || window.location.pathname + window.location.search;
+        const [urlPath, urlParams] = action.split("?");
         if (urlParams) {
-            for (const keyValue of urlParams.split('&')) {
-                const [key, value] = keyValue.split('=');
-                if (value && key !== 'search') {
+            for (const keyValue of urlParams.split("&")) {
+                const [key, value] = keyValue.split("=");
+                if (value && key !== "search") {
                     // Decode URI parameters: revert + to space then decodeURIComponent.
-                    this.options[decodeURIComponent(key.replace(/\+/g, '%20'))] = decodeURIComponent(value.replace(/\+/g, '%20'));
+                    this.options[decodeURIComponent(key.replace(/\+/g, "%20"))] = decodeURIComponent(value.replace(/\+/g, "%20"));
                 }
             }
         }
-        const pathParts = urlPath.split('/');
+        const pathParts = urlPath.split("/");
         for (const index in pathParts) {
             const value = decodeURIComponent(pathParts[index]);
-            if (index > 0 && /-[0-9]+$/.test(value)) { // is sluggish
-                this.options[decodeURIComponent(pathParts[index - 1])] = value;
+            const indexNumber = parseInt(index);
+            if (indexNumber > 0 && /-[0-9]+$/.test(value)) { // is sluggish
+                this.options[decodeURIComponent(pathParts[indexNumber - 1])] = value;
             }
         }
+    }
 
-        if (this.$input.data('noFuzzy')) {
-            $("<input type='hidden' name='noFuzzy' value='true'/>").appendTo(this.$input);
+    start() {
+        if (this.inputEl.dataset.noFuzzy) {
+            const noFuzzyEl = document.createElement("input");
+            noFuzzyEl.setAttribute("type", "hidden");
+            noFuzzyEl.setAttribute("name", "noFuzzy");
+            noFuzzyEl.setAttribute("value", "true");
+            this.insert(noFuzzyEl, this.inputEl);
+            this.registerCleanup(() => { noFuzzyEl.remove(); });
```

Comment:
oups sorry, just noticed that the code for insert is doing that:
```js
    insert(el, locationEl = this.el, position = "beforeend") {
        locationEl.insertAdjacentElement(position, el);
        this.registerCleanup(() => el.remove());
    }
```
forgot about that